seismicpro/src/gather/gather.py

```diff
@@ -1028,6 +1030,52 @@ def crop(self, origins, crop_shape, n_crops=1, stride=None, pad_mode='constant',
         origins = make_origins(origins, self.shape, crop_shape, n_crops, stride)
         return CroppedGather(self, origins, crop_shape, pad_mode, **kwargs)
 
+    @batch_method(target="t")
+    def bandpass_filter(self, n, low=None, high=None, window='hamm', **kwargs):
+        """ docs """
+        fs = 1000 / self.sample_rate
+        if low is not None and high is not None:
+            kernel = scipy.signal.firwin(n, [low, high], window=window, fs=fs, pass_zero='bandpass', **kwargs)
+        elif low is not None:
+            kernel = scipy.signal.firwin(n, low, window=window, fs=fs, pass_zero='highpass', **kwargs)
+        elif high is not None:
+            kernel = scipy.signal.firwin(n, high, window=window, fs=fs, pass_zero='lowpass', **kwargs)
+        else:
+            raise ValueError('At least one of low and high must be provided')
+
+        self.data = cv2.filter2D(self.data, ddepth=-1, kernel=kernel.reshape(1, -1))
+        return self
+
+    @batch_method(target="f")
+    def resample(self, new_sample_rate, kind=3, anti_aliasing=True):
+        """ if type(kind) == int, perform piecewise polynomial interpolation with polynomial degree = kind
+            if type(kind) == str, deligate interpolation to scipy.interpolate.intep1d.
+        """
+        current_sample_rate = self.sample_rate
+
+        # in case new sample rate becomes more, i.e. performing downsample,
+        # anti-aliasing filter is applied to preserve signal frequencies
+        if new_sample_rate > current_sample_rate and anti_aliasing:
+            nyquist_frequency = 1000 / (2 * new_sample_rate)
+
+            # estimate parameters for filter
+            n = int(10000 / (nyquist_frequency * current_sample_rate))
+            cutoff_ratio = 0.8
+
+            # perform filerting
+            self.bandpass_filter(n, high=cutoff_ratio * nyquist_frequency)
```

Comment:
```suggestion
            # Smoothly attenuate frequencies starting from 0.8 of the new Nyquist frequency so that all frequencies
            # above are zeroed out
            nyquist_frequency = 1000 / (2 * new_sample_rate)
            filter_size = int(40 * new_sample_rate / current_sample_rate)
            self.bandpass_filter(high=0.9 * nyquist_frequency, filter_size=filter_size, window="hann")
```
Mimic paradigm algorithm